doc/foresight.rst

```diff
@@ -163,109 +190,152 @@ technologies.
+myopic, perfect).
 
-General myopic code structure
----------------------------------
+**Unified configuration structure:**
+
+.. code:: yaml
 
-The myopic code solves the network for the time steps included in
-``planning_horizons`` in a recursive loop, so that:
+  co2_budget:
+    emissions_scope: All greenhouse gases - (CO2 equivalent)
+    values: fraction  # "fraction" (% of 1990 baseline) or "absolute" (Gt CO2/year)
+    upper: null       # null | scalar | {year: value}
+    lower: null       # null | scalar | {year: value}
 
-1. The existing capacities (those installed before the base year are added as
-   fixed capacities with p_nom=value, p_nom_extendable=False). E.g. for
-   baseyear=2020, capacities installed before 2020 are added. In addition, the
-   network comprises additional generator, storage, and link capacities with
-   p_nom_extendable=True. The non-solved network is saved in
-   ``resources/run_name/networks``.
+**Constraint semantics:**
 
-The base year is the first element in ``planning_horizons``. Step 1 is
-implemented with the rule add_baseyear for the base year and with the rule
-add_brownfield for the remaining planning_horizons.
+- ``upper``/``lower`` as a **dict**: apply constraints only for explicitly listed years
+  (omitted years are unconstrained). In perfect foresight, these constraints are attached
+  to the corresponding investment period. Use ``<year>: null`` to explicitly clear an
+  inherited default bound for a specific year.
+- ``upper``/``lower`` as a **scalar**: add a single timeless CO₂ constraint (no investment
+  period reference).
 
-2. The 2020 network is optimized. The solved network is saved in
-   ``results/run_name/networks``
+  - **overnight**: one CO₂ constraint for the solved model.
+  - **myopic**: one CO₂ constraint for each solved horizon.
+  - **perfect**: one joint CO₂ constraint across all horizons, added in the final horizon.
 
-3. For the next planning horizon, e.g. 2030, the capacities from a previous time
-   step are added if they are still in operation (i.e., if they fulfil planning
-   horizon <= commissioned year + lifetime). In addition, the network comprises
-   additional generator, storage, and link capacities with
-   p_nom_extendable=True. The non-solved network is saved in
-   ``results/run_name/networks``.
+**Example 1: Per-period caps with fraction values**
 
-Steps 2 and 3 are solved recursively for all the planning_horizons included in
-``config/config.yaml``.
+.. code:: yaml
 
-Rule overview
---------------
+  co2_budget:
+    values: fraction
+    upper:
+      2030: 0.450  # 45% of 1990 emissions
+      2040: 0.100
+      2050: 0.000
+
+**Example 2: Scalar cap (absolute values)**
+
+.. code:: yaml
+
+  co2_budget:
+    values: absolute
+    upper: 2.04  # Gt CO2/year, same semantics across foresight modes (see above)
+
+**Example 3: Selective year constraints**
 
-- rule add_existing baseyear
```

Comment:
Was the documentation below here for `add_existing_baseyear` (the function remains) moved somewhere? Please do not just remove this.